code/game/machinery/vendors/vending.dm

```diff
@@ -862,6 +949,133 @@
 
 /obj/machinery/economy/vending/onTransitZ()
 	return
+
+/obj/machinery/economy/vending/proc/choose_crit()
+	if(!length(possible_crits))
+		return
+	var/crit_path_selected = pick(possible_crits)
+	return GLOB.vendor_crits[crit_path_selected]
+
+/**
+ * Tilts the machine onto the atom passed in.
+ *
+ * Arguments:
+ * * victim - The thing the machine is falling on top of
+ * * crit - if true, some special damage effects might happen.
+ * * from_combat - If true, hold off on some of the additional damage and extra effects.
+ */
+/obj/machinery/economy/vending/proc/tilt(atom/victim, crit = FALSE, from_combat = FALSE)
+	if(QDELETED(src) || !has_gravity(src) || !tiltable || tilted)
+		return
+
+	visible_message("<span class='danger'>[src] tips over!</span>", "<span class='danger'>You hear a loud crash!</span>")
+	tilted = TRUE
+	layer = ABOVE_MOB_LAYER
+
+	var/should_throw_at_target = TRUE
+
+	. = FALSE
+
+	if(in_range(victim, src))
+		for(var/mob/living/L in get_turf(victim))
+			var/mob/living/carbon/C = L
+
+			var/crit_rebate = 0
+			var/damage_to_deal = squish_damage
+			if(crit || !from_combat)
+				// double the damage on a crit
+				damage_to_deal *= crit_damage_factor
+			if(HAS_TRAIT(target, TRAIT_DWARF))
```

Comment:
```suggestion
			if(HAS_TRAIT(C, TRAIT_DWARF))
```
contrary to the linter saying line 924, it is actually this use of `target` that is causing issues.